### PR TITLE
Add CIL's ptranal for `EvalFunvar` queries

### PR DESCRIPTION
**Pull request by vesalvojdani**
This will add the option to use Cil's Ptranal module to deal with dynamic function calls. This would avoid having to run the base analysis for fast analyses. The problem is that the Ptranal module's points to queries are really literal, so `*fp` (which is how function pointer calls are normalized by CIL) does not point to anything, you have to remove the dererefence and query `fp`, and then for cases like calling through pointers within structs, the analysis is flow-insensitive and that causes some issues.

- [ ] This is currently implemented as a wrapper / unit analysis to answer the queries. Should decide if this rather belongs properly to the base functionality of the framework. Depends on how much code is still needed to handle things.
- [ ] The expressions fed to Ptranal is still not very robust: there are dead code on some bench repo examples.
- [ ] Filter compatible function calls since the pointer analysis is flow-insensitive.

### CODE REVIEW

File: src/analyses/ptranalEvalFunvar.ml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    [ctx.local, ctx.local]
+
+  let combine_env ctx lval fexp f args fc au f_ask =
+    au
+
+  let combine_assign ctx (lval:lval option) fexp (f:fundec) (args:exp list) fc (au:D.t) (f_ask: Queries.ask) : D.t =
+    ctx.local
+
+  let special ctx (lval: lval option) (f:varinfo) (arglist:exp list) : D.t =
+    ctx.local
```
Comment (sim642) [inline src/analyses/ptranalEvalFunvar.ml:38]:
```ocaml
include UnitAnalysis.Spec
```

**Comment by michael-schwarz** [comment]
What are our plans with this?

**Comment by sim642** [comment]
Since Goblint CIL still has `Ptranal` and this Goblint "analysis" is very simple, there's no reason to not have this around for some modular/base-less analysis experiments.

**Comment by michael-schwarz** [comment]
> Since Goblint CIL still has Ptranal and this Goblint "analysis" is very simple, there's no reason to not have this around for some modular/base-less analysis experiments.

I think in this case it would be a good idea to address the open remarks and then merge this.

**Comment by michael-schwarz** [comment]
I have now brought this up-to-date with master. Should we merge it as-is or address the comments above before?

**Comment by michael-schwarz** [comment]
The failing indentation is already on master.

**Review by sim642 — APPROVED** [review]
I made a few small changes to make this merge-ready.

**Pull request merged**